src/yki/handler/quarantine.clj

```diff
@@ -25,29 +41,35 @@
       (POST "/" request
         :body [quarantine ::ys/quarantine-type]
         :return ::ys/response
-        (when-let [created (quarantine-db/create-quarantine! db quarantine)]
-          (audit-log/log {:request   request
-                          :target-kv {:k audit-log/quarantine
-                                      :v (:id created)}
-                          :change    {:type audit-log/create-op
-                                      :new  quarantine}})
-          (ok {:success true})))
+        (let [created (->> quarantine
+                           (with-birthdate-from-ssn)
+                           (quarantine-db/create-quarantine! db))]
+          (when created
+            (audit-log/log {:request   request
+                            :target-kv {:k audit-log/quarantine
+                                        :v (:id created)}
+                            :change    {:type audit-log/create-op
+                                        :new  created}})
+            (ok {:success true}))))
       (PUT "/:id" request
         :path-params [id :- ::ys/id]
         :body [quarantine ::ys/quarantine-type]
         (if-let [existing (quarantine-db/get-quarantine db id)]
-          (if-let [updated (quarantine-db/update-quarantine! db id quarantine)]
-            (do
-              (audit-log/log {:request   request
-                              :target-kv {:k audit-log/quarantine
-                                          :v id}
-                              :change    {:type audit-log/update-op
-                                          :old  existing
-                                          :new  updated}})
-              (ok {:success true}))
-            (do
-              (log/error "Update failed for existing quarantine with id" id)
-              (bad-request)))
+          (let [updated (->> quarantine
+                             with-birthdate-from-ssn
```

Comment:
Tässä pitäis olla kai sulut ympärillä

Reply:
Toimii ilman niitä sulkujakin, kun kerran kyseessä on yhden argumentin funktio: https://clojure.org/guides/threading_macros#thread-last

Reply:
Ehkä toi voi konsistenssin puolesta olla sulkujen kanssa, kun kerran useimmissa paikoissa sitä muotoa on tullut viljeltyä.